dcpy/models/product/metadata.py

```diff
@@ -113,19 +122,43 @@ class OrgMetadata(SortedSerializedBase, extra="forbid"):
     root_path: Path
     template_vars: dict = Field(default_factory=dict)
     metadata: OrgMetadataFile
+    column_defaults: dict[tuple[str, str], DatasetColumn]
+
+    @classmethod
+    def get_yml(cls, path: Path):
+        if path.exists():
+            with open(path, "r", encoding="utf-8") as raw:
+                return yaml.safe_load(raw)
+        else:
+            return None
+
+    @classmethod
+    def get_string_snippets(cls, path: Path) -> dict:
+        yml = cls.get_yml(path / "snippets" / "strings.yml") or {}
+        if not isinstance(yml, dict):
+            raise ValueError("snippets must be valid yml dict, not array")
+        return yml
+
+    @classmethod
+    def get_column_defaults(cls, path: Path) -> dict[tuple[str, str], DatasetColumn]:
+        yml = cls.get_yml(path / "snippets" / "column_defaults.yml")
+        columns = TypeAdapter(list[DatasetColumn]).validate_python(yml or [])
```

Comment:
This is a neat approach. I could see wanting a JSON schema for the snippets file, which might mean an actual SnippetsFile class. But this seems good for now